src/iterators.jl

```diff
@@ -83,9 +83,83 @@ function reinit!(cc::CellCache, i::Int)
     return cc
 end
 
+###############
+## FaceCache ##
+###############
+
+"""
+    FaceCache(grid::Grid, topology::ExclusiveTopology)
+    FaceCache(dh::AbstractDofHandler, topology::ExclusiveTopology)
+
+Create a cache object with pre-allocated memory for the coordinates and facets of a face in the face skeleton.
+The cache is updated for a new cell by calling `reinit!(cache, this_face, neighbor_face = nothing)` where
+`this_face::FaceIndex` and `neighbor_face::FaceIndex` are the interface facets.
+
+**Struct fields of `FaceCache`**
+ - `fc.this_coords :: Vector{<:Vec}`: current cell node coordinates
+ - `fc.neighbor_coords :: Vector{<:Vec}`: neighbor cell node coordinates
+ - `fc.this_face :: Vector{<:Vec}`: local face index for current cell
+ - `fc.neighbor_face :: Vector{<:Vec}`: local face index for neighbor cell
+ - `fc.orientation_info :: InterfaceOrientationInfo`: whether the neighbor orientation info relative to current face
+ - `fc.grid :: AbstractGrid`: grid information used in iteration
+ - `fc.topology :: ExclusiveTopology`: topology information used in iteration
+
+**Methods with `FaceCache`**
+ - `reinit!(cache::FaceCache, this_face::FaceIndex, neighbor_face::FaceIndex)`: reinitialize [`FaceCache`](@ref)
+
+See also [`FaceIterator`](@ref).
+"""
+struct FaceCache{X,G<:AbstractGrid}
```

Comment:
Although this case is more complicated, you could potentially re-use some of the re-use from `CellCache` as done in #495

Reply:
Sure, thanks!